internal/pkg/cli/env_delete.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+// DeleteEnvOpts holds the fields needed to delete an environment.
+type DeleteEnvOpts struct {
+	Env              string
+	SkipConfirmation bool
+}
+
+// Ask is a no-op for this command.
+func (opts *DeleteEnvOpts) Ask() error {
+	return nil
+}
+
+// Validate returns an error if the environment name does not exist in the project, or there are existing applications
+// in the environment, or the environment is not available using the EnvironmentManagerRole.
+func (opts *DeleteEnvOpts) Validate() error {
+	return nil
+}
+
+// Execute deletes the environment from the project by first deleting the stack and then removing the entry from the store.
+func (opts *DeleteEnvOpts) Execute() error {
+	return nil
+}
+
+// RecommendedActions is a no-op for this command.
+func (opts *DeleteEnvOpts) RecommendedActions() []string {
+	return nil
+}
+
+// BuildEnvDeleteCmd builds the command to delete environment(s).
+func BuildEnvDeleteCmd() *cobra.Command {
+	opts := &DeleteEnvOpts{}
+	cmd := &cobra.Command{
+		Use:   "delete [name]",
+		Short: "Deletes an environment from your project.",
+		Example: `
+  Delete the "test" environment.
+  /code $ archer env delete test
+
+  Delete the "test" environment without prompting.
+  /code $ archer env delete test prod --yes`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				fmt.Print(cmd.UsageString())
+				return nil
+			}
+			if args[0] == "help" {
+				fmt.Print(cmd.UsageString())
+				return nil
+			}
```

Comment:
Is the intent here to support `env delete help` and print usage? If so, I think we need to remove the possibility of naming an environment `help` first or at the same time as supporting this.

Reply:
#387

Reply:
Wrapped our commands with a `runCmdE` function to make sure that we print the help menu if a user enters help as an argument. Let me know what y'all think.

I think the nice part of this approach is that we can also catch the errors returned from the `Run` methods and possibly display something more human friendly